roles/jupyterhub/templates/jupyterhub_config.py

```diff
@@ -94,7 +94,7 @@ def sync_users(username, min_uid=1_000_000, max_uid=1_000_000_000):
 
         payload = user
         payload['attributes']['jupyterhubCheck'] = 'true'
-        payload['attributes']['homeDirectory'] = f'/home/{username}'
+        payload['attributes']['homeDirectory'] = f"/home/{user['username']}"
 
         initial_uid = (hash(user['username']) % (max_uid - min_uid)) + min_uid
         while initial_uid in current_uids:
```

Comment:
Side question: I'm also still a bit uncertain whether the code below this is correct. Is it not possible that the `user` in this loop already has a UID recorded (i.e. `user.get('attributes', {}).get('uidNumber')` is present), and this will end up changing it?